File: 3. Linked List/9_longest_streak.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

def longest_streak(head):
    current = head
    max_streak = 0
    current_streak = 0
    prev_val = None

    while current is not None:
        if prev_val == current.val:
            current_streak += 1
        else:
            current_streak = 1
        if current_streak > max_streak:
            max_streak = current_streak
        prev_val = current.val
        current = current.next
    return max_streak

File: 3. Linked List/9_longest_streak/9_longest_streak.java
class Node<T> {
    T val;
    Node<T> next;

    public Node(T val) {
        this.val = val;
        this.next = null;
    }
}

class Source {
    public static <T> int longestStreak(Node<T> head) {
        Node<T> current = head;
        int maxStreak = 0;
        int currentStreak = 0;
        T prevVal = null;

        while (current != null) {
            if (prevVal == current.val) {
                currentStreak++;
            } else {
                currentStreak = 1;
            }
            if (currentStreak > maxStreak) {
                maxStreak = currentStreak;
            }
            prevVal = current.val;
            current = current.next;
        }
        return maxStreak;
    }
}
